synapse/storage/databases/main/relations.py

```diff
@@ -91,7 +91,7 @@ def __init__(
 
         self._msc3440_enabled = hs.config.experimental.msc3440_enabled
 
-    @cached(tree=True)
+    @cached(num_args=9, tree=True)
```

Comment:
Without this the cache does nothing (since `EventBase` instances are never equal).

Reply:
I don't think it would be that hard to make `@cached` accept a `cached_param_names`  parameter fwiw. The `num_args` basically just gets converted into a list of args to cache:

https://github.com/matrix-org/synapse/blob/26211fec24d8d0a967de33147e148166359ec8cb/synapse/util/caches/descriptors.py#L91-L106

Reply:
Meaning so that we define something like `def f(event_id, event, blah = None, foo = True)` but cache on `event_id, blah, foo`?

Reply:
Although you'd need to make those parameters incompatible and the descriptors are already fairly complicated... hmm.... I think it is probably OK since not passing it is a programming error?

Reply:
See #12189 for being able to skip args.